gn3/db/rdf/wiki.py

```diff
@@ -0,0 +1,168 @@
+        query, WIKI_CONTEXT,
+        sparql_uri
+    )
+    data = results.get("data")
+    if not data:
+        return results
+    return results
+
+
+def get_comment_history(comment_id: int, sparql_uri: str) -> dict:
+    """Get all the historical data for a given id"""
+    query = Template("""
+$prefix
+
+CONSTRUCT {
+    ?uid rdfs:label ?symbolName ;
+         gnt:reason ?reason ;
+         gnt:species ?species ;
+         dct:references ?pmid ;
+         foaf:homepage ?weburl ;
+         rdfs:comment ?comment ;
+         foaf:mbox ?email ;
+         gnt:initial ?usercode ;
+         gnt:belongsToCategory ?category ;
+         gnt:hasVersion ?versionId ;
+         dct:created ?created .
+} WHERE {
+    ?symbolId rdfs:label ?symbolName .
+    ?uid rdf:type gnc:GNWikiEntry ;
+         rdfs:comment ?comment ;
+         gnt:symbol ?symbolId ;
+         dct:created ?createTime ;
+         dct:hasVersion ?version ;
+         dct:identifier $comment_id ;
+         dct:identifier ?id_ .
+    OPTIONAL { ?uid gnt:reason ?reason_ } .
+    OPTIONAL {
+        ?uid gnt:belongsToSpecies ?speciesId .
+        ?speciesId gnt:shortName ?species_ .
+    } .
+    OPTIONAL { ?uid dct:references ?pmid . } .
+    OPTIONAL { ?uid foaf:homepage ?weburl_ . } .
+    OPTIONAL { ?uid gnt:initial ?usercode_ . } .
+    OPTIONAL { ?uid foaf:mbox ?email_ . } .
+    OPTIONAL { ?uid gnt:belongsToCategory ?category_ . } .
+    BIND (str(?version) AS ?versionId) .
+    BIND (str(?createTime) AS ?created) .
+    BIND (COALESCE(?reason_, "") AS ?reason) .
+    BIND (COALESCE(?weburl_, "") AS ?weburl) .
+    BIND (COALESCE(?usercode_, "") AS ?usercode) .
+    BIND (COALESCE(?email_, "") AS ?email) .
+    BIND (COALESCE(?species_, "") AS ?species) .
+    BIND (COALESCE(?category_, "") AS ?category) .
+}
+""").substitute(prefix=RDF_PREFIXES, comment_id=comment_id)
+    results = query_frame_and_compact(
+        query, WIKI_CONTEXT,
+        sparql_uri
+    )
+    data = results.get("data")
+    for result in data:
+        categories = result.get("categories") or []
+        if categories and isinstance(categories, str):
+            result["categories"] = [categories]
+        pmids = result.get("pubmed_ids")
+        if pmids and isinstance(pmids, str):
+            result["pubmed_ids"] = [pmids]
+        elif pmids:
+            result["pubmed_ids"] = [int(pmid) for pmid in pmids]
+        else:
+            result["pubmed_ids"] = []
+        result["version"] = int(result["version"])
+
+    # We manually sort the array, since the SPARQL engine does not
+    # provide a guarantee that it will support an ORDER BY clause in a
+    # CONSTRUCT. Using ORDER BY on a solution sequence for a CONSTRUCT
+    # or DESCRIBE query has no direct effect because only SELECT
+    # returns a sequence of results.  See:
+    # <https://stackoverflow.com/questions/78186393/how-apply-an-order-to-construct-in-sparql#:~:text=SPARQL%20engine%20does%20respect%20such,is%20a%20set%20of%20triples.>
```

Comment:
can we replace this with: https://stackoverflow.com/q/78186393

Reply:
Done.